content/switching-to-v0-mangling-on-nightly.md

```diff
@@ -0,0 +1,171 @@
++++
+path = "2025/10/28/switching-to-v0-mangling-on-nightly"
+title = "Switching to Rust's own mangling scheme on nightly"
+authors = ["David Wood"]
+
+[extra]
+team = "the compiler team"
+team_url = "https://www.rust-lang.org/governance/teams/compiler"
++++
+
+**TL;DR:** rustc will use its own "v0" mangling scheme by default on nightly
+versions
+
+#### Context
+
+When Rust is compiled into object files and binaries, each item (functions,
+statics, etc) must have a globally unique  "symbol" identifying it.
+
+In C, the symbol name of a function is just the name that the function was
+defined with, such as `strcmp`. This is straightforward and easy to
+understand, but requires that each item have a globally unique name
+that don't overlap with any symbols from shared libraries that might be linked
+against. If two items had the same symbol then when the linker tried to resolve
+a symbol to an address in memory (of a function, say), then it wouldn't know
+which symbol is the correct one.
+
+Languages like Rust and C++ define "symbol mangling schemes", leveraging information
+from the type system to give each item a unique symbol name. Otherwise every
+instantiation of a generic or templated function (or an overload in C++), which has
+the same name in the surface language would end up with clashing symbols.
```

Comment:
Feels like this should mention paths vs identifiers, e.g. Rust lets you have `a::foo` and `b::foo`. It's not just generics.